modules/Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java

```diff
@@ -449,37 +473,74 @@ private boolean inclineCraft(HashHitBox hitBox){
             MovecraftLocation nl = ml.translate(dx, dy, dz);
             if (hitBox.contains(nl))
                 continue;
-            if (ml.getY() > hitBox.getMinY())
-                continue;
             collisionBox.add(nl);
         }
+
+        int elevation = 0;
         for (MovecraftLocation ml : collisionBox){
-            if (!ml.toBukkit(craft.getW()).getBlock().getType().equals(Material.AIR) && !craft.getType().getPassthroughBlocks().contains(ml.toBukkit(craft.getW()).getBlock().getType()) && !craft.getType().getHarvestBlocks().contains(ml.toBukkit(craft.getW()).getBlock().getType())){
-                return true;
+            Material testType = ml.toBukkit(craft.getW()).getBlock().getType();
+            if (testType == Material.AIR || craft.getType().getPassthroughBlocks().contains(testType)) {
+                continue;
+            }
+            MovecraftLocation surfaceLoc = surfaceLoc(ml);
+            if (elevation < surfaceLoc.subtract(ml).getY()) {
+                elevation = surfaceLoc.subtract(ml).getY();
             }
+
+        }
+        if (elevation == 0) {
+            return false;
+        }
+        HashHitBox movedCollBox = new HashHitBox();
+        for (MovecraftLocation ml : collisionBox) {
+            movedCollBox.add(ml.translate(0, elevation, 0));
+
+        }
+        climbDistance = movedCollBox.getMinY() - hitBox.getMinY();
+        for (MovecraftLocation ml : movedCollBox) {
+            Material test = ml.toBukkit(craft.getW()).getBlock().getType();
+            if (test != Material.AIR && !craft.getType().getPassthroughBlocks().contains(test)) {
+                continue;
+            }
+            return true;
         }
 
         return false;
     }
 
-    private boolean isOnGround(HashHitBox hitBox){
+    private int dropDistance (HashHitBox hitBox) {
         MutableHitBox bottomLocs = new HashHitBox();
-        MutableHitBox translatedBottomLocs = new HashHitBox();
         for (MovecraftLocation location : hitBox){
-            MovecraftLocation beneath = location.translate(0, -1, 0);
-            //If the hitbox contains the location beneath this one, continue the loop
-            if (hitBox.contains(beneath)){
-                    continue;
+            if (location.getY() != hitBox.getLocalMinY(location.getX(), location.getZ())) {
+                continue;
             }
-            //If not, check if it is an exterior location
-            int y;
-            for (y = beneath.getY() ; y >= hitBox.getMinY() ; y--){
-                if (hitBox.contains(beneath.getX(), y, beneath.getZ())){
+            //Otherwise, add to bottom locations
+            bottomLocs.add(location);
+        }
+        int dropDistance = 0;
+        Material testType = null;
```

Comment:
Don't initialize to `null` if not necessary.

Reply:
It is necessary due to that it is assigned a value in a for loop inside a do while after this line. If I remove the null initialization, the IDE will display an error